processor/elasticprocessor/internal/hostmetrics/disk.go

```diff
@@ -0,0 +1,152 @@
+package hostmetrics
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func addDiskMetrics(metrics pmetric.MetricSlice, resource pcommon.Resource, dataset string) error {
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+		if metric.Name() == "system.disk.io" || metric.Name() == "system.disk.operations" {
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+
+				var device string
+				if d, ok := dp.Attributes().Get("device"); ok {
+					device = d.Str()
+				} else {
+					continue
+				}
+
+				if direction, ok := dp.Attributes().Get("direction"); ok {
+					name := metric.Name()
+					timestamp := dp.Timestamp()
+					value := dp.IntValue()
+
+					switch direction.Str() {
+					case "read":
+						addDiskIntMetric(metrics, resource, dataset, name, device, "read", timestamp, value)
+					case "write":
+						addDiskIntMetric(metrics, resource, dataset, name, device, "write", timestamp, value)
+					}
+				}
+			}
+		} else if metric.Name() == "system.disk.operation_time" {
+			var multiplier float64
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+
+				var device string
+				if d, ok := dp.Attributes().Get("device"); ok {
+					device = d.Str()
+				} else {
+					continue
+				}
+				if direction, ok := dp.Attributes().Get("direction"); ok {
+					name := metric.Name()
+					timestamp := dp.Timestamp()
+					value := dp.DoubleValue()
+					if name == "system.disk.operation_time" {
```

Comment:
this `if` is redundant

Reply:
To be future-ready in case we get a new metric ?

Reply:
that suggests a code smell to me. there are a couple of others in this file which make me think this code could do with some simplification - for example, the fact we have a parameter which in some cases is the 'direction', and in others a static string `io`. in addition we have two functions with very similar names `addDiskDoubleMetric` and `addDiskIntMetric`, which i might suggest we combine using type parameters, but then when you look deeper, there is non-type-specific logic in there (like the multiplier).